### ISSUE TITLE
Goldmark Error After Updating

### ISSUE THREAD

**Issue by sethjensenbrown** [gohugoio/hugo#8733]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version 0.84.4

</pre>

### Does this issue reproduce with the latest release?

Yes.

### Issue

After updating from v0.82.1 to 0.83.0 and up I receive the following error when running `hugo` or `hugo server`. I am okay for now using the 0.82.1 image for CI, but would like to be able to stay up to date. Any idea what might be causing the problem or what the solution might be?

```
Start building sites … 
hugo v0.84.4+extended darwin/amd64 BuildDate=unknown
goroutine 336 [running]:
runtime/debug.Stack(0x5f1f300, 0x6b2be20, 0xc000949880)
	runtime/debug/stack.go:24 +0x9f
github.com/gohugoio/hugo/markup/goldmark.(*goldmarkConverter).Convert.func1(0xc000846480, 0xc001fb2a80, 0x8e1, 0x9c4, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/gohugoio/hugo/markup/goldmark/convert.go:239 +0x213
panic(0x59ea6c0, 0xc000b7e930)
	runtime/panic.go:965 +0x1b9
github.com/yuin/goldmark/text.(*Segment).Value(0xc00286f1e0, 0xc001fb2a80, 0x8e1, 0x9c4, 0x2a, 0x0, 0x0)
	github.com/yuin/goldmark@v1.3.8/text/segment.go:48 +0x3f6
github.com/yuin/goldmark-highlighting.(*HTMLRenderer).renderFencedCodeBlock(0xc0005b9c70, 0x5f181a8, 0xc00072e240, 0xc001fb2a80, 0x8e1, 0x9c4, 0x5f2d7a0, 0xc001ec92c0, 0x901, 0x3, ...)
	github.com/yuin/goldmark-highlighting@v0.0.0-20200307114337-60d527fdb691/highlighting.go:465 +0x927
github.com/yuin/goldmark/renderer.(*renderer).Render.func2(0x5f2d7a0, 0xc001ec92c0, 0x1, 0x3, 0x0, 0x0)
	github.com/yuin/goldmark@v1.3.8/renderer/renderer.go:166 +0x108
github.com/yuin/goldmark/ast.walkHelper(0x5f2d7a0, 0xc001ec92c0, 0xc00286f688, 0x3, 0x0, 0x0)
	github.com/yuin/goldmark@v1.3.8/ast/ast.go:492 +0x43
github.com/yuin/goldmark/ast.walkHelper(0x5f2de60, 0xc001fd5680, 0xc00286f688, 0x3, 0x0, 0x0)
	github.com/yuin/goldmark@v1.3.8/ast/ast.go:498 +0x16b
github.com/yuin/goldmark/ast.walkHelper(0x5f2dd40, 0xc001fd53b0, 0xc00286f688, 0x3, 0x0, 0x0)
	github.com/yuin/goldmark@v1.3.8/ast/ast.go:498 +0x16b
github.com/yuin/goldmark/ast.walkHelper(0x5f2de60, 0xc001fd5320, 0xc00286f688, 0x3, 0x0, 0x0)
	github.com/yuin/goldmark@v1.3.8/ast/ast.go:498 +0x16b
github.com/yuin/goldmark/ast.walkHelper(0x5f2dd40, 0xc001fd5290, 0xc00286f688, 0x3, 0x0, 0x0)
	github.com/yuin/goldmark@v1.3.8/ast/ast.go:498 +0x16b
github.com/yuin/goldmark/ast.walkHelper(0x5f2d560, 0xc001fd4480, 0xc00286f688, 0x5f181a8, 0xc00072e240, 0x1)
	github.com/yuin/goldmark@v1.3.8/ast/ast.go:498 +0x16b
github.com/yuin/goldmark/ast.Walk(...)
	github.com/yuin/goldmark@v1.3.8/ast/ast.go:487
github.com/yuin/goldmark/renderer.(*renderer).Render(0xc00010e500, 0x5ee9f60, 0xc00072e240, 0xc001fb2a80, 0x8e1, 0x9c4, 0x5f2d560, 0xc001fd4480, 0x0, 0x0)
	github.com/yuin/goldmark@v1.3.8/renderer/renderer.go:161 +0x13c
github.com/gohugoio/hugo/markup/goldmark.(*goldmarkConverter).Convert(0xc000846480, 0xc001fb2a80, 0x8e1, 0x9c4, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/gohugoio/hugo/markup/goldmark/convert.go:265 +0x5b9
github.com/gohugoio/hugo/hugolib.(*pageContentOutput).renderContentWithConverter(0xc0013a7dc0, 0x5efb4b0, 0xc000846480, 0xc001fb2a80, 0x8e1, 0x9c4, 0x4e9d701, 0xc0013a7e94, 0xc00286fa30, 0x4eca7c0, ...)
	github.com/gohugoio/hugo/hugolib/page__per_output.go:378 +0xff
github.com/gohugoio/hugo/hugolib.(*pageContentOutput).renderContent(0xc0013a7dc0, 0xc001fb2a80, 0x8e1, 0x9c4, 0xc000563401, 0x0, 0x5efb5c8, 0xc000af9860, 0xffffffffffffffff)
	github.com/gohugoio/hugo/hugolib/page__per_output.go:374 +0x71
github.com/gohugoio/hugo/hugolib.newPageContentOutput.func1(0x0, 0x0)
	github.com/gohugoio/hugo/hugolib/page__per_output.go:126 +0xb05
github.com/gohugoio/hugo/hugolib.newPageContentOutput.func2(0x5f0ce80, 0xc001fd0960, 0x0, 0x0, 0x0, 0x0)
	github.com/gohugoio/hugo/hugolib/page__per_output.go:205 +0x26
github.com/gohugoio/hugo/lazy.(*Init).withTimeout.func1(0xc001337220, 0x5f0ce80, 0xc001fd0960, 0xc001fd09c0)
	github.com/gohugoio/hugo/lazy/init.go:176 +0x47
created by github.com/gohugoio/hugo/lazy.(*Init).withTimeout
	github.com/gohugoio/hugo/lazy/init.go:175 +0x110
Total in 1146 ms
Error: Error building site: "/Users/sethjensenbrown/Documents/projects/developer-manual/content/docs/frontend/angular2/pagination-component.md:1:1": [BUG] goldmark: runtime error: slice bounds out of range [1932:1929]: create an issue on GitHub attaching the file in: /var/folders/k4/r4dj9jtj51vc5ns_vfxgl82h0000gq/T/hugo_bugs/goldmark_f5131dd03ac7ff8c9406d416677e3c86.txt
```

Here is the file it asked me to attach:

[goldmark_f5131dd03ac7ff8c9406d416677e3c86.txt](https://github.com/gohugoio/hugo/files/6780022/goldmark_f5131dd03ac7ff8c9406d416677e3c86.txt)

**Comment by jmooring** [comment]
I can reproduce the problem with v0.84.4.
The problem was fixed v0.85.0.

**Comment by sethjensenbrown** [comment]
Oh wow, I didn't realize there was a new version already! My apologies!  v0.85.0 fixed the issue for me as well, thanks!
